src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -2510,17 +2528,19 @@ class StubGenerator: public StubCodeGenerator {
              tmpU = isLU ? tmp2 : tmp1, // where to keep U for comparison
              tmpL = isLU ? tmp1 : tmp2; // where to keep L for comparison
 
-    // make sure main loop is 8 byte-aligned, we should load another 4 bytes from strL
-    // cnt2 is >= 68 here, no need to check it for >= 0
-    __ lwu(tmpL, Address(strL));
-    __ addi(strL, strL, wordSize / 2);
-    __ ld(tmpU, Address(strU));
-    __ addi(strU, strU, wordSize);
-    __ inflate_lo32(tmp3, tmpL);
-    __ mv(tmpL, tmp3);
-    __ xorr(tmp3, tmpU, tmpL);
-    __ bnez(tmp3, CALCULATE_DIFFERENCE);
-    __ addi(cnt2, cnt2, -wordSize / 2);
+    if ((base_offset1 % 8) == 0) {
```

Comment:
The condition should be `((base_offset1 % 8) != 0)`? Seems we try to make the `strL` aligned in the main loop below.

Reply:
And do we need below piece of code when `AvoidUnalignedAccesses == false`?

Reply:
The alignment is 8 bytes without COH. And the caller of the stub has loaded 4 characters [1]. So when we need to process another 4 characters to make `strL` 8-byte aligned when entering the main loop. But this is not needed with COH, because the alignment is 4 bytes. That's why I added this if check. Hope that explains.

And I have added check for `AvoidUnalignedAccesses` to this logic.

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/stubGenerator_riscv.cpp#L2511